memory/boat_manager.py

```diff
@@ -0,0 +1,85 @@
+from engine.mathlib import Vec3
+from memory.core import mem_handle
+
+
+class BoatManager:
+    def __init__(self):
+        self.memory = mem_handle()
+        self.base = None
+        self.position = Vec3(None, None, None)
+        self.rotation_x = None
+        self.rotation_y = None
+        self.speed = 0
+
+    def update(self):
+        if self.memory.ready_for_updates:
+            try:
+                if self.base is None:
+                    singleton_ptr = self.memory.get_singleton_by_class_name(
+                        "BoatManager"
+                    )
+                    if singleton_ptr is None:
+                        return
+
+                    self.base = self.memory.get_class_base(singleton_ptr)
+                    if self.base == 0x0:
+                        return
+
+                else:
+                    # Update fields
+                    self._read_position()
+                    self._read_rotation()
+                    self._read_speed()
+            except Exception as _e:
+                # logger.debug(f"BoatManager Reloading {type(_e)}")
+                self.__init__()
+
+    def _read_position(self):
+        if self.memory.ready_for_updates:
+            # (BoatInstance) k__BackingField -> boatController -> currentTargetPosition
+            ptr = self.memory.follow_pointer(self.base, [0x40, 0x40, 0x84])
+            if ptr:
+                x = self.memory.read_float(ptr + 0x0)
+                y = self.memory.read_float(ptr + 0x4)
+                z = self.memory.read_float(ptr + 0x8)
+
+                self.position = Vec3(x, y, z)
+                return
+
+        self.position = Vec3(None, None, None)
+
+    def _read_rotation(self):
+        if self.memory.ready_for_updates:
+            # (BoatInstance) k__BackingField -> boatSnapRotation -> pitchRollLocalRotation
```

Comment:
So to be specific, are these part of a quaternion? Or are they pitchRoll Euler angles? What is the unit, degrees or radians?

Reply:
its set as a UnityEngine.Core.Quaternion in the game code, but the fields dont make a whole lot of sense as i'm not sure how to parse what its trying to do. The docs say "dont touch this" basically so i kind of just take the values that changed.